src/client/java/daos-java-api/src/main/native/com_intel_daos_client_DaosFsClient.c

```diff
@@ -0,0 +1,911 @@
+/*
+ * (C) Copyright 2018-2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#include "com_intel_daos_client_DaosFsClient.h"
+#include <sys/stat.h>
+#include <sys/xattr.h>
+#include <dirent.h>
+#include <gurt/common.h>
+#include <libgen.h>
+#include <stdio.h>
+#include <daos.h>
+#include <daos_fs.h>
+#include <daos_jni_common.h>
+#include <fcntl.h>
+
+static jclass daos_io_exception_class;
+
+static jmethodID new_exception_msg;
+static jmethodID new_exception_cause;
+static jmethodID new_exception_msg_code_msg;
+static jmethodID new_exception_msg_code_cause;
+
+static int ERROR_PATH_LEN = 256;
+static int ERROR_NOT_EXIST = 2;
+static int ERROR_LOOKUP_MAX_RETRIES = 100;
+
+jint JNI_OnLoad(JavaVM* vm, void *reserved) {
+    JNIEnv *env;
+    if ((*vm)->GetEnv(vm, (void**)&env, JNI_VERSION) != JNI_OK) {
```

Comment:
(style)  Missing a blank line after declarations
(style)  "(foo**)" should be "(foo **)"